lib/libc/misc/lib_dumpbuffer.c

```diff
@@ -67,6 +67,10 @@
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+char assert_info_str[CONFIG_STDIO_BUFFER_SIZE] = {'\0', };
```

Comment:
I think it would be better to move this declaration to some file related to debug. 
Best place would be `lib/libc/misc/lib_dbg.c`. But, this file is not built always. It depends on `CONFIG_DEBUG` macro. 
So, maybe we can move it to `lib/libc/syslog/lib_syslog.c`